File: frontend/src/features/analytics/types/analytics.ts
export interface CampaignAnalytics {
  campaign_id: string;
  sent: number;
  delivered: number;
  opened: number;
  clicked: number;
  bounced: number;
  complained: number;
  unsubscribed: number;
  open_rate: number;
  click_rate: number;
  bounce_rate: number;
  complaint_rate: number;
  unsubscribe_rate: number;
}

export interface DashboardAnalytics {
  total_contacts: number;
  active_contacts: number;
  total_campaigns: number;
  total_sent: number;
  total_opens: number;
  total_clicks: number;
  avg_open_rate: number;
  avg_click_rate: number;
}

export interface AnalyticsTimeSeriesPoint {
  timestamp: string;
  value: number;
}

export interface CampaignPerformanceItem {
  campaign_id: string;
  campaign_name: string;
  sent: number;
  open_rate: number;
  click_rate: number;
  created_at: string;
}

export interface TopCampaignsResponse {
  items: CampaignPerformanceItem[];
}

export interface AnalyticsExportResponse {
  download_url: string;
}

export interface SendProgress {
  total: number;
  sent: number;
  status: string;
}

File: backend/app/schemas/analytics.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class CampaignAnalyticsResponse(BaseModel):
    campaign_id: UUID
    sent: int
    delivered: int
    opened: int
    clicked: int
    bounced: int
    complained: int
    unsubscribed: int
    open_rate: float
    click_rate: float
    bounce_rate: float
    complaint_rate: float
    unsubscribe_rate: float


class DashboardAnalyticsResponse(BaseModel):
    total_contacts: int
    active_contacts: int
    total_campaigns: int
    total_sent: int
    total_opens: int
    total_clicks: int
    avg_open_rate: float
    avg_click_rate: float


class AnalyticsTimeSeriesPoint(BaseModel):
    timestamp: datetime
    value: int


class CampaignPerformanceItem(BaseModel):
    campaign_id: UUID
    campaign_name: str
    sent: int
    open_rate: float
    click_rate: float
    created_at: datetime


class TopCampaignsResponse(BaseModel):
    items: list[CampaignPerformanceItem]


class AnalyticsExportResponse(BaseModel):
    download_url: str
